core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -67,27 +69,250 @@ export class ReportGenerationService {
+   * This function iterates over each operator, retrieves its detailed results via `retrieveOperatorInfoReport`,
+   * and then compiles these results into a final HTML document. The report is then generated and made available for download.
+   *
+   * @param {Object} payload - An object containing all necessary data to process and generate the report.
+   * @param {string[]} payload.operatorId - An array of operator IDs representing each operator in the workflow.
+   * @param {string} payload.workflowSnapshotURL - A URL pointing to a snapshot of the current workflow, to be included in the report.
+   * @param {string} payload.workflowName - The name of the workflow, used to generate a relevant filename for the report.
+   *
+   * @returns {void} - This function does not return a value. It processes the operators and triggers the generation of an HTML report.
+   */
+
+  public getAllOperatorResults(payload: {
+    operatorId: string[]; // Changed from `operators: OperatorDetail[]` to `operatorIds: string[]` as requested
+    workflowSnapshotURL: string;
+    workflowName: string;
+  }): void {
+    const allResults: { operatorId: string; html: string }[] = [];
+
+    const promises = payload.operatorId.map(operatorId => {
+      return this.retrieveOperatorInfoReport(operatorId, operatorId, allResults); // Updated to pass only operatorId
+    });
+
+    Promise.all(promises)
+      .then(() => {
+        const sortedResults = payload.operatorId.map(
+          id => allResults.find(result => result.operatorId === id)?.html || ""
+        );
+        this.generateReportAsHtml(payload.workflowSnapshotURL, sortedResults, payload.workflowName);
+      })
+      .catch(error => {
+        this.notificationService.error("Error in generating operator results: " + error.message);
+      });
+  }
+
+  /**
+   * Retrieves and processes detailed results for a specified operator, generating a structured HTML representation.
+   * This function covers multiple cases, including handling paginated results, snapshot data, and scenarios where
+   * no results are found for the operator. The resulting HTML content is stored in the `allResults` array.
+   *
+   * @param {string} operatorId - The unique identifier of the operator for which results are being processed.
+   * @param {any} operatorInfo - Metadata and configuration details of the operator, utilized to render a JSON editor within the HTML.
+   * @param {Array<{operatorId: string, html: string}>} allResults - An array used to collect HTML content for each operator's processed result.
+   *
+   * @returns {Promise<void>} - A promise that resolves once the operator's results have been processed and the HTML has been appended to `allResults`.
+   */
+
+  public retrieveOperatorInfoReport(
+    operatorId: string,
+    operatorInfo: any,
+    allResults: { operatorId: string; html: string }[]
+  ): Promise<void> {
```

Comment:
can it return a Promise<{operaetorId: string; html: string}[]>? Also remove the operatorInfo parameter